Large_unary test between node and ruby fail

node-node pass

ruby-ruby fail
onnadionne@donnadionne0:~/grpc-git/grpc/tools/gce_setup$ grpc_interop_test large_unary grpc-docker-testclients ruby grpc-docker-server ruby
will run:
  bash -l -c "sudo docker run grpc/ruby bin/bash -l -c 'ruby /var/local/git/grpc/src/ruby/bin/interop/interop_client.rb --use_test_ca --use_tls --server_host_override=foo.test.google.fr --server_host=10.240.248.89 --server_port=8060 --test_case=large_unary'"
on grpc-docker-testclients
Warning: Permanently added '130.211.252.68' (ECDSA) to the list of known hosts.
Using /usr/local/rvm/gems/ruby-2.1.5
stdin: is not a tty
I0221 22:30:45.979717547       1 census_init.c:41] Initialize census library.
I0221 22:30:45.979897262       1 census_tracing.c:157] Initialize census trace store.
I0221 22:30:45.980448729       1 census_rpc_stats.c:225] Initialize census stats store.
/var/local/git/grpc/src/ruby/bin/interop/interop_client.rb:59:in `<main>': uninitialized constant Google::RPC (NameError)
I0221 22:30:46.231500795       1 census_init.c:47] Shutdown census library.
I0221 22:30:46.231585956       1 census_rpc_stats.c:238] Shutdown census stats store.
I0221 22:30:46.232002792       1 census_tracing.c:170] Shutdown census trace store.
ERROR: (gcloud.compute.ssh) [/usr/bin/ssh] exited with return code [1].

ruby-node fail
donnadionne@donnadionne0:~/grpc-git/grpc/tools/gce_setup$ grpc_interop_test large_unary grpc-docker-testclients ruby grpc-docker-server node
will run:
  bash -l -c "sudo docker run grpc/ruby bin/bash -l -c 'ruby /var/local/git/grpc/src/ruby/bin/interop/interop_client.rb --use_test_ca --use_tls --server_host_override=foo.test.google.fr --server_host=10.240.248.89 --server_port=8040 --test_case=large_unary'"
on grpc-docker-testclients
Warning: Permanently added '130.211.252.68' (ECDSA) to the list of known hosts.
Using /usr/local/rvm/gems/ruby-2.1.5
stdin: is not a tty
I0221 22:31:19.742610801       1 census_init.c:41] Initialize census library.
I0221 22:31:19.742766648       1 census_tracing.c:157] Initialize census trace store.
I0221 22:31:19.743329732       1 census_rpc_stats.c:225] Initialize census stats store.
/var/local/git/grpc/src/ruby/bin/interop/interop_client.rb:59:in `<main>': uninitialized constant Google::RPC (NameError)
I0221 22:31:19.987486156       1 census_init.c:47] Shutdown census library.
I0221 22:31:19.987564463       1 census_rpc_stats.c:238] Shutdown census stats store.
I0221 22:31:19.987837827       1 census_tracing.c:170] Shutdown census trace store.
ERROR: (gcloud.compute.ssh) [/usr/bin/ssh] exited with return code [1].

node-ruby fail
donnadionne@donnadionne0:~/grpc-git/grpc/tools/gce_setup$ grpc_interop_test large_unary grpc-docker-testclients node grpc-docker-server ruby

will run:
  bash -l -c "sudo docker run grpc/node /usr/bin/nodejs /var/local/git/grpc/src/node/interop/interop_client.js --use_tls=true --use_test_ca=true --server_host_override=foo.test.google.fr --server_host=10.240.248.89 --server_port=8060 --test_case=large_unary"
on grpc-docker-testclients
Warning: Permanently added '130.211.252.68' (ECDSA) to the list of known hosts.
I0221 22:31:56.659741961       1 census_init.c:41] Initialize census library.
I0221 22:31:56.660000135       1 census_tracing.c:157] Initialize census trace store.
I0221 22:31:56.660758565       1 census_rpc_stats.c:225] Initialize census stats store.
I0221 22:31:56.690200790      10 resolve_address.c:186] logspam: getaddrinfo(10.240.248.89, 8060) resolved 1 addrs in 0ms:
I0221 22:31:56.690333591      10 resolve_address.c:192] logspam:   [0] 10.240.248.89:8060
I0221 22:31:56.762578147       8 ssl_transport_security.c:166]      HANDSHAKE START -  before/connect initialization  - UNKWN 
I0221 22:31:56.762794196       8 ssl_transport_security.c:166]                 LOOP -  before/connect initialization  - UNKWN 
I0221 22:31:56.763274616       8 ssl_transport_security.c:166]                 LOOP -     SSLv3 write client hello A  - 3WCH_A
I0221 22:31:56.862517015       8 ssl_transport_security.c:166]                 LOOP -      SSLv3 read server hello A  - 3RSH_A
I0221 22:31:56.863017529       8 ssl_transport_security.c:166]                 LOOP - SSLv3 read server certificate   - 3RSC_A
I0221 22:31:56.863304839       8 ssl_transport_security.c:166]                 LOOP - SSLv3 read server key exchange  - 3RSKEA
I0221 22:31:56.863425285       8 ssl_transport_security.c:166]                 LOOP -       SSLv3 read server done A  - 3RSD_A
I0221 22:31:56.863992261       8 ssl_transport_security.c:166]                 LOOP - SSLv3 write client key exchang  - 3WCKEA
I0221 22:31:56.864241812       8 ssl_transport_security.c:166]                 LOOP - SSLv3 write change cipher spec  - 3WCCSA
I0221 22:31:56.864325209       8 ssl_transport_security.c:166]                 LOOP -         SSLv3 write finished A  - 3WFINA
I0221 22:31:56.864436133       8 ssl_transport_security.c:166]                 LOOP -               SSLv3 flush data  - 3FLUSH
I0221 22:31:56.962493449       8 ssl_transport_security.c:166]                 LOOP - SSLv3 read server session tick  - UNKWN 
I0221 22:31:56.962664214       8 ssl_transport_security.c:166]                 LOOP -          SSLv3 read finished A  - 3RFINA
I0221 22:31:56.963005295       8 ssl_transport_security.c:166]       HANDSHAKE DONE - SSL negotiation finished succe  - SSLOK 
D0221 22:31:56.966994402      13 frame_settings.c:219] CHTTP2: got setting 4 = 65535

assert.js:326
assert.ifError = function(err) { if (err) {throw err;}};
                                                 ^
AssertionError: null === 0
    at /var/local/git/grpc/src/node/interop/interop_client.js:86:12
    at GrpcClientStream.forwardData (/var/local/git/grpc/src/node/src/surface_client.js:177:9)
    at GrpcClientStream.emit (events.js:95:17)
    at GrpcClientStream.<anonymous> (_stream_readable.js:765:14)
    at GrpcClientStream.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:427:10)
    at emitReadable (_stream_readable.js:423:5)
    at readableAddChunk (_stream_readable.js:166:9)
    at GrpcClientStream.Readable.push (_stream_readable.js:128:10)
    at readCallback (/var/local/git/grpc/src/node/src/client.js:108:14)

--- a/src/node/interop/messages.proto
+++ b/src/node/interop/messages.proto
@@ -49,7 +49,7 @@ enum PayloadType {
 // A block of data, to simply increase gRPC message size.
 message Payload {
   // The type of data in body.
-  optional PayloadType type = 1;
+  optional PayloadType type = 1 [default = COMPRESSABLE];
   // Primary contents of payload.
   optional bytes body = 2;
 }
@@ -58,7 +58,7 @@ message Payload {
 message SimpleRequest {
   // Desired payload type in the response from the server.
   // If response_type is RANDOM, server randomly chooses one from other formats.
-  optional PayloadType response_type = 1;
+  optional PayloadType response_type = 1 [default = COMPRESSABLE];
 
   // Desired payload size in the response from the server.
   // If response_type is COMPRESSABLE, this denotes the size before compression.
@@ -116,7 +116,7 @@ message StreamingOutputCallRequest {
   // If response_type is RANDOM, the payload from each response in the stream
   // might be of different types. This is to simulate a mixed type of payload
   // stream.
-  optional PayloadType response_type = 1;
+  optional PayloadType response_type = 1 [default = COMPRESSABLE];
 
   // Configuration for each expected response message.
   repeated ResponseParameters response_parameters = 2;

